CMakeLists.txt

```diff
@@ -257,11 +285,21 @@ endif()
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
+
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+if (NOT CMAKE_CROSSCOMPILING)
+  find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
```

Comment:
Perhaps in this situation we can force CMake to use Armadillo in header-only mode?  It would be great to use the system-installed version, since by itself Armadillo is header-only.

Reply:
I know, I agree totally with you, but unfortunately as detailed above we can not use `find_package()` since it is going to search in the system header.

Reply:
Ok, let's work out the reply above and we can then return to this with the conclusion. :+1:

Reply:
For this (and other comments like it), I think we can use `find_package()` just fine, expecting either that the user has (a) set `ARMADILLO_INCLUDE_DIR` correctly, (b) installed Armadillo to the root of their cross-compilation toolchain include directory, (c) used the auto-downloader, which can set `ARMADILLO_INCLUDE_DIR` before the `find_package()` call.

Reply:
I will give it a try, it should work.

Reply:
@rcurtin I test it locally with ensmallen since it doesn't have `if(NOT CMAKE_CROSSCOMPILING)` before `find_package()`.
It can be possible to remove the `if(NOT CMAKE_CROSSCOMPILING)` but there are two cases:
1) if ensmallen is installed manually, then it is fine, since it is installed in `/usr/local/include`
2) if ensmallen is installed via system package manager, then it will be found in `/usr/include` which will show the above issue.
3) if ensmallen is not found, it will be downloaded automatically which is also fine too.

This will be the same issue with `stb`.

Therefore, in order to avoid being in case 2) I have to put `if(NOT CMAKE_CROSSCOMPILING)` before `find_package` for the two above cases.

Anythoughts?

Reply:
Does your cross-compilation setup set `CMAKE_FIND_ROOT_PATH` to something other than the default?  If `CMAKE_FIND_ROOT_PATH` is set correctly, then I believe `/usr/include/` should not be searched.

Reply:
Yes, `CMAKE_FIND_ROOT_PATH`, is correctly set, I find the issue it is coming from `CMAKE_SYSROOT`. 
Even I set it to empty, it has a default value inside, in order to change the default value, I have set the `SYSROOT` to the same value of `FIND_ROOT_PATH`.
I removed all these `if(NOT CMAKE_CROSSCOMPILING)`. It is working fine now. :+1: